src/uct/api/uct.h

```diff
@@ -414,6 +414,7 @@ enum {
 typedef enum {
     UCT_MD_MEM_TYPE_HOST = 0,      /**< Default system memory */
     UCT_MD_MEM_TYPE_CUDA,          /**< NVIDIA CUDA memory */
+    UCT_MD_MEM_TYPE_CUDA_MANAGED,  /**< NVIDIA CUDA managed memory */
```

Comment:
do we really need this new element? Can't we just treat it as HOST mem type?

Reply:
@brminich We can't treat it as HOST mem type because it can lead to registering managed memory and potentially performing RDMA on it. We don't want that to happen because pages concerning the buffer region may not be updated with the contents of the GPU. This is because page migration is on demand.

For example, calling MPI_Send(managed_buffer, ...) immediately after a kernel call is valid for the MPI program but treating the managed buffer as host will lead to erroneous data being transferred. At least this is the case on non-p9 architectures. We need to explicitly copy these buffers or touch them in some way from the host before transferring them. Please let me know if I'm being clear. 

@evgeny-leksikov The preference for 'managed memory' over 'unified memory' is because the memory is allocated using cudaMallocManaged. Unified memory is too generic a term. Don't you think?

Reply:
Just thought if we can create some abstraction here.. don't mind it.

Reply:
At least I would suggest to document somewhere managed == unified. Maybe in comments.

Reply:
@evgeny-leksikov @shamisp I'll add a comment now.